### PR TITLE
BACKLOG-22448 Added a better check for the presence of graalvm

### PR DESCRIPTION
**Pull request by Fgerthoffert**
https://jira.jahia.org/browse/BACKLOG-22448
 
- Check done using the `org.graalvm.version` property
- Added a check on the version of GraalVM (22.3.3+ required)

### CODE REVIEW

File: src/main/java/org/jahia/modules/sam/healthcheck/probes/SupportedStackJSModulesProbe.java [hunk 1]
```diff
@@ -25,18 +25,21 @@ public ProbeStatus getStatus() {
...
+            status = ProbeStatusUtils.aggregateStatus(status, String.format("GraalVM not detected on the environment (detected vendor: %s with JVM version: %s), after switching to GraalVM 22.3.3+ with JVM version 17+, make sure to enable the Javascript extension", vmVendor, jvmVersion), ProbeStatus.Health.RED);
+        } else { 
+            if (jvmVersion.compareTo(new Version("22.3.3")) <= 0) {
+                status = ProbeStatusUtils.aggregateStatus(status, String.format("GraalVM version 22.3.3 or newer required (detected GraalVM version: %s)", graalVMVersion), ProbeStatus.Health.RED);
             }
             if (jvmVersion.compareTo(new Version("17")) <= 0) {
-                status = ProbeStatusUtils.aggregateStatus(status, String.format("GraalVM with JVM version 17 or newer required (detected: %s)", jvmVersion), ProbeStatus.Health.RED);
+                status = ProbeStatusUtils.aggregateStatus(status, String.format("GraalVM with JVM version 17 or newer required (detected JVM version: %s)", jvmVersion), ProbeStatus.Health.RED);
             }            
             if (vmVendor.contains("GraalVM") && !isJavaScriptModuleInstalled()) {
```
Comment (jkevan) [inline src/main/java/org/jahia/modules/sam/healthcheck/probes/SupportedStackJSModulesProbe.java:44]: I see we are still testing for: `vmVendor.contains("GraalVM")` should we consider refactor also this check ?

File: src/main/java/org/jahia/modules/sam/healthcheck/probes/SupportedStackJSModulesProbe.java [hunk 2]
```diff
@@ -25,18 +25,21 @@ public ProbeStatus getStatus() {
...
         ProbeStatus status = new ProbeStatus("No issues to report", ProbeStatus.Health.GREEN);
         if (!isNpmModulesEngineStarted) {
             status = ProbeStatusUtils.aggregateStatus(status, "The environment is not running JS modules (npm-modules-engine stopped or not present)", ProbeStatus.Health.GREEN);
-        } else {
-            if (!vmVendor.contains("GraalVM")) {
-                status = ProbeStatusUtils.aggregateStatus(status, String.format("GraalVM not detected on the environment (detected vendor: %s), after switching to GraalVM make sure to enable the Javascript extension", vmVendor), ProbeStatus.Health.RED);
+        } else if (System.getProperty("org.graalvm.version", "Unknown") == "Unknown") {
+            status = ProbeStatusUtils.aggregateStatus(status, String.format("GraalVM not detected on the environment (detected vendor: %s with JVM version: %s), after switching to GraalVM 22.3.3+ with JVM version 17+, make sure to enable the Javascript extension", vmVendor, jvmVersion), ProbeStatus.Health.RED);
+        } else { 
+            if (jvmVersion.compareTo(new Version("22.3.3")) <= 0) {
```
Comment (jkevan) [inline src/main/java/org/jahia/modules/sam/healthcheck/probes/SupportedStackJSModulesProbe.java:38]: Something is wrong here. the if statement is checking: `jvmVersion.compareTo(new Version("22.3.3")`
I think it should be: `graalVMVersion.compareTo(new Version("22.3.3")`

File: src/main/java/org/jahia/modules/sam/healthcheck/probes/SupportedStackJSModulesProbe.java [hunk 3]
```diff
@@ -25,18 +25,21 @@ public ProbeStatus getStatus() {
...
 
         // This probe is only relevant for Jahia 8.2.0.0+ in which npm-modules-engine is available. 
         // Not testing Jahia version since it is not to be backported to older versions of SAM.
         ProbeStatus status = new ProbeStatus("No issues to report", ProbeStatus.Health.GREEN);
         if (!isNpmModulesEngineStarted) {
             status = ProbeStatusUtils.aggregateStatus(status, "The environment is not running JS modules (npm-modules-engine stopped or not present)", ProbeStatus.Health.GREEN);
-        } else {
-            if (!vmVendor.contains("GraalVM")) {
-                status = ProbeStatusUtils.aggregateStatus(status, String.format("GraalVM not detected on the environment (detected vendor: %s), after switching to GraalVM make sure to enable the Javascript extension", vmVendor), ProbeStatus.Health.RED);
+        } else if (System.getProperty("org.graalvm.version", "Unknown") == "Unknown") {
```
Comment (jkevan) [inline src/main/java/org/jahia/modules/sam/healthcheck/probes/SupportedStackJSModulesProbe.java:35]: `System.getProperty("org.graalvm.version", "Unknown")`
is already called at like 28, may be we could introduce a new variable and reuse it.
Also no need for default value if you only want to check if a sysrtem property exists, we can check for null directly.

**Review by jkevan — CHANGES_REQUESTED** [review]

File: src/main/java/org/jahia/modules/sam/healthcheck/probes/SupportedStackJSModulesProbe.java (hunk 1 above)
Comment (Fgerthoffert) [reply src/main/java/org/jahia/modules/sam/healthcheck/probes/SupportedStackJSModulesProbe.java:44]: I believe this one is correct., there are two different version, JDK and GraalVM

**Review by jkevan — DISMISSED** [review]

**Pull request merged**